web/services/nft_service.py

```
@@ -0,0 +1,243 @@
+        self.contract = self.web3.eth.contract(address=self.contract_address, abi=self.contract_abi)
+
+    def _load_contract_abi(self):
+        """Load contract ABI from file"""
+        try:
+            with open("web/static/contracts/nft_abi.json", "r") as f:
+                return json.load(f)
+        except FileNotFoundError:
+            raise FileNotFoundError("❌ ABI file not found!")
+
+    @staticmethod
+    def upload_to_ipfs(file_path, max_retries=3):
```

Ruff (0.8.2)

Missing return type annotation for staticmethod `upload_to_ipfs`

Add return type annotation: `Optional[str]`

(ANN205)